### PR TITLE
Add invokeInstance to gRPC API v2

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Add invokeInstance to gRPC API v2.
Depends on https://github.com/Concordium/concordium-grpc-api/pull/16

## Changes

_Describe the changes that were needed.

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 1]
```diff
@@ -1648,6 +1699,83 @@ impl ConsensusContainer {
...
+        let amount = request.amount.as_ref().require()?.value;
+
+        let (receive_name_ptr, receive_name_len) =
+            crate::grpc2::types::receive_name_to_ffi(request.entrypoint.as_ref().require()?)
+                .require()?;
+
+        // Parameter to ffi
+        let (parameter_ptr, parameter_len) = {
+            let bytes = &request.parameter.as_ref().require()?.value;
+            (bytes.as_ptr(), bytes.len().try_into().map_err(|_| tonic::Status::internal(""))?)
```
Comment (Bargsteen) [inline concordium-node/src/consensus_ffi/ffi.rs:1745]: Please write an error message even though it won't happen.

File: concordium-node/src/consensus_ffi/ffi.rs [hunk 2]
```diff
@@ -1648,6 +1699,83 @@ impl ConsensusContainer {
...
+        // Optional Address to ffi
+        let (
+            invoker_address_type,
+            invoker_account_address_ptr,
+            invoker_contract_index,
+            invoker_contract_subindex,
+        ) = if let Some(address) = &request.invoker {
+            match address.r#type.as_ref().require()? {
+                crate::grpc2::types::address::Type::Account(account) => {
+                    (1, crate::grpc2::types::account_address_to_ffi(&account).require()?, 0, 0)
```
Comment (Bargsteen) [inline concordium-node/src/consensus_ffi/ffi.rs:1726]: Fix clippy error:

```suggestion
                    (1, crate::grpc2::types::account_address_to_ffi(account).require()?, 0, 0)
```

**Review by Bargsteen — APPROVED** [review]
LGTM 👍 There is just a single clippy error, but I've suggested the fix.

File: concordium-consensus/src/Concordium/GRPC2.hs [hunk 3]
```diff
@@ -1302,6 +1307,19 @@ instance ToProto Energy where
   type Output Energy = Proto.Energy
   toProto = mkWord64
 
+instance ToProto InvokeContract.InvokeContractResult where
+  type Output InvokeContract.InvokeContractResult = Either ConversionError Proto.InvokeInstanceResponse
```
[the remark this answers is not in the document]
Comment (abizjak) [reply concordium-consensus/src/Concordium/GRPC2.hs:1314]: I've added a comment.

**Pull request merged**